### ISSUE TITLE
Wrong recognition of the VLC on the Android TV

### ISSUE THREAD

**Issue by valib** [UniversalMediaServer/UniversalMediaServer#2285]
@Subjunk I think that we have a problem to properly recognise the VLC running on the Android TV. It is reported in http://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=14422. At the beginning is the TV recognised properly as the Sony Bravia TV but when user starts the VLC their requests are still recognised as Sony Bravia TV requests even if  the VLC sends the different USER-AGENT like `USER-AGENT: Linux/3.10.79, UPnP/1.0, Portable SDK for UPnP devices/1.6.19` and different UUID. So the renderer capability is different and causes problems. Maybe this is the USER-AGENT Regex setting problem in the `VLC-for-desktop.conf` or the problem is in the code.

**Comment by Nadahar** [comment]
IP addresses only are used for consecutive matches, to speed things up I assume. So, once a renderer is "recognized" on a certain IP, UMS won't even try to re-evaluate this - which is a logical fallacy for any device that can run more than one "renderer software". This goes for computers, mobile phones, Android TVs etc. I pointed this out many years ago.

The fact is that it's a bit tricky since there's no way to match UPnP discoveries to HTTP header discoveries except via IP. So, if you turned off the whole "one device one IP" logic, most renderers would be detected twice - and UMS wouldn't be able to tell that they were in fact the same renderer.

**Comment by SubJunk** [comment]
Yeah I'm not sure there is anything we can do about this, it's more a limitation of DLNA itself, isn't it? Please close the issue if you agree, or suggest a way to improve it

**Comment by Nadahar** [comment]
I don't quite know where to "place the blame". UPnP AV never intended to use User-Agent for "renderer detection", so that in itself is somewhat of a "hack". Much more responsibility was supposed to be with the renderer, so the server wasn't intended to have to "adapt" to the renderer - which would mean that identifying the renderer isn't important.

That model doesn't quite work in the real-world though, as most renderer implementations aren't that robust, which is why things have developed as they have I guess. It's still a mess though, it would have been extremely much easier if UPnP AV mandated that the renderer UUID was sent as a HTTP header for example.
